scene/2d/tile_map.cpp

```diff
@@ -3172,23 +3294,11 @@ void TileMap::set_y_sort_enabled(bool p_enable) {
 		return;
 	}
 	Node2D::set_y_sort_enabled(p_enable);
-	_clear_internals();
-	_recreate_internals();
-	emit_signal(CoreStringNames::get_singleton()->changed);
-	update_configuration_warnings();
-}
-
-void TileMap::_clear_internals() {
-	// Clear quadrants.
-	for (Ref<TileMapLayer> &layer : layers) {
-		layer->clear_internals();
-	}
-}
-
-void TileMap::_recreate_internals() {
 	for (Ref<TileMapLayer> &layer : layers) {
-		layer->recreate_internals();
+		layer->notify_tile_map_change(TileMapLayer::DIRTY_FLAGS_TILE_MAP_Y_SORT_ENABLED);
 	}
+	emit_signal(CoreStringNames::get_singleton()->changed);
+	update_configuration_warnings();
 }
 
 /////////////////////////////// Rendering //////////////////////////////////////
```

Comment:
Probably unrelated to this PR, but the methods below don't look like rendering related.